### DISCUSSION TITLE
Invalid characters in the name of a note

### DISCUSSION THREAD

**Discussion by louwelyn-jeremy** [SilentVoid13/Templater#1328] [Help]
# Things I have tried
I use this template to create or rename a note and move it to a specific folder :
```
<%* let qcFileName = await tp.system.prompt("name for the note",tp.file.title);
await tp.file.rename(qcFileName);
-%>
some texte here
<%*await tp.file.move("/KMS_jeremy/GTD/GTD-meeting/" + qcFileName)-%>
```

But I have a problem with for example video name from YouTube when the title of a video has a special character.
I found a solution [here](https://forum.obsidian.md/t/templater-invalid-characters-in-the-name-of-a-note/48346)
so I try this:
```
<%* let qcFileName = await tp.system.prompt("name for the note",tp.file.title);
qcFileName = qcFileName.replace(/[^\w\s]/gi, ‘_’);
await tp.file.rename(qcFileName);
-%>
some texte here
<%*await tp.file.move("/KMS_jeremy/GTD/GTD-meeting/" + qcFileName)-%>
```

But I have a parsing error --> Template syntax error: Invalid or unexpected token
Where am I wrong ?

**Comment by Zachatoo** [comment]
You have "smart quotes" instead of regular quotes. replace this line
```js
qcFileName = qcFileName.replace(/[^\w\s]/gi, ‘_’);
```
with this line.
```js
qcFileName = qcFileName.replace(/[^\w\s]/gi, '_');
```
Some OSs (I think iOS) have smart quotes enabled by default, which is annoying for coding since they aren't parsed by JavaScript.
